products/iambeta/api.yaml

```diff
@@ -94,11 +94,199 @@ objects:
+    create_url: projects/{{project}}/locations/global/workloadIdentityPools/{{workload_identity_pool_id}}/providers/?workloadIdentityPoolProviderId={{workload_identity_pool_provider_id}}
+    update_verb: :PATCH
+    update_mask: true
+    description: A configuration for an external identity provider.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Managing workload identity providers':
+          'https://cloud.google.com/iam/docs/manage-workload-identity-pools-providers#managing_workload_identity_providers'
+      api: 'https://cloud.google.com/iam/docs/reference/rest/v1beta/projects.locations.workloadIdentityPools.providers'
+    properties:
+      - !ruby/object:Api::Type::String
+         name: 'workloadIdentityPoolId'
+         description: |
+           The ID used for the pool, which is the final component of the pool resource name. This
+           value should be 4-32 characters, and may contain the characters [a-z0-9-]. The prefix
+           `gcp-` is reserved for use by Google, and may not be specified.
+         required: true
+         input: true
+         url_param_only: true
+      - !ruby/object:Api::Type::String
+         name: 'workloadIdentityPoolProviderId'
+         description: |
+           The ID for the provider, which becomes the final component of the resource name. This
+           value must be 4-32 characters, and may contain the characters [a-z0-9-]. The prefix
+           `gcp-` is reserved for use by Google, and may not be specified.
+         required: true
+         input: true
+         url_param_only: true
+      - !ruby/object:Api::Type::Enum
+        name: 'state'
+        description: |
+          The state of the provider.
+          STATE_UNSPECIFIED: State unspecified.
+          ACTIVE: The provider is active, and may be used to validate authentication credentials.
+          DELETED: The provider is soft-deleted. Soft-deleted providers are permanently deleted
+          after approximately 30 days. You can restore a soft-deleted provider using
+          UndeleteWorkloadIdentityPoolProvider. You cannot reuse the ID of a soft-deleted provider
+          until it is permanently deleted.
+        output: true
+        values:
+          - :STATE_UNSPECIFIED
+          - :ACTIVE
+          - :DELETED
+      - !ruby/object:Api::Type::String
+        name: 'displayName'
+        description: A display name for the provider. Cannot exceed 32 characters.
+      - !ruby/object:Api::Type::String
+        name: 'description'
+        description: A description for the provider. Cannot exceed 256 characters.
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          The resource name of the provider as
+          `projects/<projectnumber>/locations/global/workloadIdentityPools/<pool-id>/providers/<provider-id>`.
```

Comment:
```suggestion
          `projects/{project}/locations/global/workloadIdentityPools/{workload_identity_pool_id}/providers/{workload_identity_pool_provider_id}`.
```
Suggest we align to one naming convention here. Name, Self-Link and id all have different formats even though they are identical references